### PR TITLE
Allow setting minimum window size

### PR DESCRIPTION
**Pull request by Finnerale**
This implements setting the minimum window size (#632) **for Linux**.
The minimum size is also passed to the Mac and Windows WindowBuilder but they are not being utilized.

I've added this to the `calc` example and `TODO` tags to the implementation missing places as it was done in #587.

### CODE REVIEW

File: druid/examples/calc.rs [hunk 1]
```diff
@@ -200,9 +200,9 @@ fn build_calc() -> impl Widget<CalcState> {
 }
 
 fn main() {
-    let window = WindowDesc::new(build_calc).title(
-        LocalizedString::new("calc-demo-window-title").with_placeholder("Simple Calculator"),
-    );
+    let window = WindowDesc::new(build_calc)
+        .title(LocalizedString::new("calc-demo-window-title").with_placeholder("Simple Calculator"))
+        .with_min_size((250.0, 350.0));
```
Comment (cmyr) [inline druid/examples/calc.rs]: can we move this to another example? I'd like to make calc be non-resizable (see #683)

Comment (Finnerale) [reply druid/examples/calc.rs]: Sure, maybe the flex example?

Comment (cmyr) [reply druid/examples/calc.rs]: I like that idea!

File: druid/examples/flex.rs [hunk 2]
```diff
@@ -323,7 +323,8 @@ fn make_ui() -> impl Widget<AppState> {
 
 fn main() -> Result<(), PlatformError> {
     let main_window = WindowDesc::new(make_ui)
-        .window_size((600., 600.00))
+        .window_size((620., 265.00))
```
Comment (cmyr) [inline druid/examples/flex.rs]:
```suggestion
        .window_size((620., 600.00))
```
I think it's clearer when these are different, and we currently can't display the  column example  at our initial size.

File: druid/src/app.rs [hunk 3]
```diff
@@ -179,6 +181,14 @@ impl<T: Data> WindowDesc<T> {
         self
     }
 
+    /// Set the minimal window size, similar to [`window_size`]
```
Comment (cmyr) [inline druid/src/app.rs]:
```suggestion
    /// Set the minimum window size.
    ///
    /// To  set the initial window size, see [`window_size`].
```

**Pull request merged**